packages/web-extension/test/messaging/remoteApi.test.ts

```diff
@@ -163,285 +166,455 @@ const setUp = (someNumbers$: Observable<bigint> = of(0n), nestedSomeNumbers$ = o
-          someNumbersSource$.next(-1n);
-          someNumbersSource$.next(0n);
-          setTimeout(() => {
-            const emittedFromSubscriptionBeforeConnect = firstValueFrom(sut.consumer.someNumbers$.pipe(toArray()));
-            sut.hostMessenger.connect();
-            setTimeout(() => {
-              const emittedFromSubscriptionAfterConnect = firstValueFrom(sut.consumer.someNumbers$.pipe(toArray()));
-              setTimeout(async () => {
-                someNumbersSource$.next(1n);
-                setTimeout(() => sut.hostSubscription.shutdown());
-                expect(await emittedFromSubscriptionBeforeConnect).toEqual([0n, 1n]);
-                expect(await emittedFromSubscriptionBeforeConnect).toEqual(await emittedFromSubscriptionAfterConnect);
-                done();
-              }, 1);
-            }, 1);
-          }, 1);
+        it('calls remote method and resolves result', async () => {
+          const resolved = sut.consumer.nested.addOneNoZero(2n);
+          sut.emitInitial();
+          expect(await resolved).toBe(3n);
+        });
+        test('requestOptions | validate', async () => {
+          await expect(() => sut.consumer.nested.addOneNoZero(0n)).rejects.toThrowError();
         });
       });
 
-      describe('mirrors source errors', () => {
-        const subscribeAndAssertError = (done: Function) => {
-          sut.consumer.someNumbers$.subscribe({
-            error: (err) => {
-              expect(err instanceof Error).toBe(true);
-              expect(err.message).toBe('err');
-              done();
-            }
-          });
-        };
+      describe('top-level property', () => {
+        beforeEach(() => sut.emitInitial());
+
+        it('calls remote method and resolves result', async () => {
+          expect(await sut.consumer.addOne(2n)).toBe(3n);
+        });
 
-        it('when thrown after subscription', (done) => {
-          subscribeAndAssertError(done);
-          someNumbersSource$.error(new Error('err'));
+        test('requestOptions | transform', async () => {
+          expect(await sut.consumer.addOneTransformedToAddTwo(1n)).toBe(3n);
         });
 
-        it('when thrown before subscription', (done) => {
-          someNumbersSource$.error(new Error('err'));
-          subscribeAndAssertError(done);
+        it('calls remote method multiple times and resolves result', async () => {
+          expect(await sut.consumer.addOne(2n)).toBe(3n);
+          expect(await sut.consumer.addOne(2n)).toBe(3n);
         });
+
+        it('calls remote method from new source and resolves result', async () => {
+          expect(await sut.consumer.addOne(2n)).toBe(3n);
+          // add 11 instead of 1 to differentiate from original api object
+          otherApi.addOne = async (arg: bigint) => arg + 11n;
+          sut.api$.next(otherApi);
+          expect(await sut.consumer.addOne(2n)).toBe(13n);
+        });
+
+        // TODO: this fails in CI
```

Comment:
Can the tests not be `skip`ped?

Reply:
They were disabled previously. It's just the diff that is making it look as if it's a new change.
I can debug why it was failing in the first place, but I would rather do it before the PR is merged, as a separate task